src/core/net.rs

```diff
@@ -945,6 +990,33 @@ mod ffi {
         }
     }
 
+    #[allow(clippy::missing_safety_doc)]
+    #[no_mangle]
+    pub unsafe extern "C" fn tcp_stream_check_connected(
+        s: *const TcpStream,
+        out_errno: *mut c_int,
+    ) -> c_int {
+        let s = s.as_ref().unwrap();
```

Comment:
Your earlier fn checked `out_errno` - is it worth adding the same check to this one?

```suggestion
        assert!(!out_errno.is_null());
        
        let s = s.as_ref().unwrap();
```